packages/react/tooltip/src/Tooltip.tsx

```diff
@@ -14,45 +13,113 @@ import { createPopperScope } from '@radix-ui/react-popper';
 import { Portal } from '@radix-ui/react-portal';
 import { Slottable } from '@radix-ui/react-slot';
 import * as VisuallyHiddenPrimitive from '@radix-ui/react-visually-hidden';
-import { useCallbackRef } from '@radix-ui/react-use-callback-ref';
 import { useId } from '@radix-ui/react-id';
-import { createStateMachine } from './createStateMachine';
-import { tooltipStateChart } from './tooltipStateChart';
 
 import type * as Radix from '@radix-ui/react-primitive';
 import type { Scope } from '@radix-ui/react-context';
 
+type ScopedProps<P = {}> = P & { __scopeTooltip?: Scope };
+const [createTooltipContext, createTooltipScope] = createContextScope('Tooltip', [
+  createPopperScope,
+]);
+const usePopperScope = createPopperScope();
```

Comment:
I wonder if these ones should stay closer to where they're used, it's less to do with the definition of the scope, but more with using it in a part that uses a scope. What do you think?

Reply:
There are multiple parts that will use this scope though... so do I just define it alongside the first one that uses it?

Reply:
Ah ok, makes sense then. 👍